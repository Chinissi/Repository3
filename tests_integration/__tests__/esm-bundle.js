"use strict";

const globby = require("globby");

const prettier = require("../../dist/esm/standalone.mjs").default;
const codeSamples = require("../../website/playground/codeSamples").default;

const parsers = [
  "angular",
  "babel",
  "css",
  "flow",
  "glimmer",
  "graphql",
  "html",
  "markdown",
  "typescript",
  "yaml",
];

const pluginsPromise = globby("../../dist/esm/parser-*.mjs", {
  cwd: __dirname,
}).then((files) => {
  return files.map((file) => require(file).default);
});

parsers.forEach((parser) => {
  test(`format ${parser}`, async () => {
    expect(
      prettier.format(codeSamples(parser), {
        parser,
        plugins: await pluginsPromise,
      })
    ).toMatchSnapshot();
  });
});
