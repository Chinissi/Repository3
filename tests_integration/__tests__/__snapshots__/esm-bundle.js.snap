// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format angular 1`] = `
"<!DOCTYPE html>
<HTML CLASS=\\"no-js mY-ClAsS\\">
  <HEAD>
    <META CHARSET=\\"utf-8\\" />
    <TITLE>My tITlE</TITLE>
    <META NAME=\\"description\\" content=\\"My CoNtEnT\\" />
  </HEAD>
  <body>
    <P>Hello world!<BR /> This is HTML5 Boilerplate.</P>
    <SCRIPT>
      window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new
      Date; ga('create', 'UA-XXXXX-Y', 'auto'); ga('send', 'pageview')
    </SCRIPT>
    <SCRIPT
      src=\\"https://www.google-analytics.com/analytics.js\\"
      ASYNC
      DEFER
    ></SCRIPT>
  </body>
</HTML>
"
`;

exports[`format babel 1`] = `
"function HelloWorld({
  greeting = \\"hello\\",
  greeted = '\\"World\\"',
  silent = false,
  onMouseOver,
}) {
  if (!greeting) {
    return null;
  }

  // TODO: Don't use random in render
  let num = Math.floor(Math.random() * 1e7)
    .toString()
    .replace(/\\\\.\\\\d+/gi, \\"\\");

  return (
    <div
      className=\\"HelloWorld\\"
      title={\`You are visitor number \${num}\`}
      onMouseOver={onMouseOver}
    >
      <strong>
        {greeting.slice(0, 1).toUpperCase() + greeting.slice(1).toLowerCase()}
      </strong>
      {greeting.endsWith(\\",\\") ? (
        \\" \\"
      ) : (
        <span style={{ color: \\"grey\\" }}>\\", \\"</span>
      )}
      <em>{greeted}</em>
      {silent ? \\".\\" : \\"!\\"}
    </div>
  );
}
"
`;

exports[`format css 1`] = `
"@media (max-width: 480px) {
  .bd-examples {
    margin-right: -0.75rem;
    margin-left: -0.75rem;
  }

  .bd-examples > [class^=\\"col-\\"] {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}
"
`;

exports[`format flow 1`] = `
"declare export function graphql<
  Props,
  Variables,
  Component: React$ComponentType<Props>
>(
  query: GQLDocument,
  config?: Config<Props, QueryConfigOptions<Variables>>
): (
  Component: Component
) => React$ComponentType<
  $Diff<
    React$ElementConfig<Component>,
    {
      data: Object | void,
      mutate: Function | void,
    }
  >
>;

declare type FetchPolicy =
  | \\"cache-first\\"
  | \\"cache-and-network\\"
  | \\"network-only\\"
  | \\"cache-only\\";
"
`;

exports[`format glimmer 1`] = `
"<div class=\\"entry\\">
  <h1>
    {{title}}
  </h1>
  <div class=\\"body\\">
    {{body}}
  </div>
</div>"
`;

exports[`format graphql 1`] = `
"query Browse(
  $offset: Int
  $limit: Int
  $categories: [String!]
  $search: String
) {
  browse(
    limit: $limit
    offset: $offset
    categories: $categories
    search: $search
  ) {
    total
    results {
      title
      price
    }
  }
}
"
`;

exports[`format html 1`] = `
"<!DOCTYPE html>
<html class=\\"no-js mY-ClAsS\\">
  <head>
    <meta charset=\\"utf-8\\" />
    <title>My tITlE</title>
    <meta name=\\"description\\" content=\\"My CoNtEnT\\" />
  </head>
  <body>
    <p>
      Hello world!<br />
      This is HTML5 Boilerplate.
    </p>
    <script>
      window.ga = function () {
        ga.q.push(arguments);
      };
      ga.q = [];
      ga.l = +new Date();
      ga(\\"create\\", \\"UA-XXXXX-Y\\", \\"auto\\");
      ga(\\"send\\", \\"pageview\\");
    </script>
    <script
      src=\\"https://www.google-analytics.com/analytics.js\\"
      async
      defer
    ></script>
  </body>
</html>
"
`;

exports[`format markdown 1`] = `
"# Header

_Look,_ code blocks are formatted _too!_

\`\`\`js
function identity(x) {
  return x;
}
\`\`\`

| Pilot    | Airport | Hours |
| -------- | :-----: | ----: |
| John Doe |   SKG   |  1338 |
| Jane Roe |   JFK   |   314 |

---

- List
- with a [link] (/to/somewhere)
- and [another one]

  [another one]: http://example.com \\"Example title\\"

Lorem ipsum dolor sit amet, consectetur adipiscing elit.
Curabitur consectetur maximus risus, sed maximus tellus tincidunt et.
"
`;

exports[`format typescript 1`] = `
"interface MyInterface {
  foo(): string;
  bar: Array<number>;
}

export abstract class Foo implements MyInterface {
  foo() {
    // TODO: return an actual value here
    return \\"hello\\";
  }
  get bar() {
    return [1, 2, 3];
  }
}

type RequestType =
  | \\"GET\\"
  | \\"HEAD\\"
  | \\"POST\\"
  | \\"PUT\\"
  | \\"OPTIONS\\"
  | \\"CONNECT\\"
  | \\"DELETE\\"
  | \\"TRACE\\";
"
`;

exports[`format yaml 1`] = `
"---
invoice: 34843
date: 2001-01-23
bill-to: &id001
  given: Chris
  family: Dumars
  address:
    lines: |
      458 Walkman Dr.
      Suite #292
    city: Royal Oak
    state: MI
    postal: 48046
ship-to: *id001
product:
  - sku: BL394D
    quantity: 4
    description: Basketball
    price: 450.00

  - sku: BL4438H
    quantity: 1
    description: Super Hoop
    price: 2392.00

tax: 251.42
total: 4443.52
comments: >
  Late afternoon is best.
  Backup contact is Nancy
  Billsmer @ 338-4338.
"
`;
